lib/Index/IndexRecord.cpp

```diff
@@ -495,6 +495,22 @@ emitDataForSwiftSerializedModule(ModuleDecl *module,
                                  IndexUnitWriter &parentUnitWriter,
                                  const PathRemapper &pathRemapper,
                                  SourceFile *initialFile) {
+  // Reload resilient modules from swiftinterface to avoid indexing
+  // internal details.
+  if (module->getResilienceStrategy() == ResilienceStrategy::Resilient) {
+    module->getASTContext().setIgnoreAdjacentModules(true);
```

Comment:
Yeah, this is an attempt at having this mode turn on only in case there are resilient dependencies. We don't need to clear the cache otherwise. It's not ideal but it may provide some small gains until we have all the compile ignore the adjacent swiftmodule file.